exec/task_config_source.go

```diff
@@ -84,15 +84,11 @@ func (configSource DeprecationConfigSource) FetchConfig(repo *worker.ArtifactRep
 		return atc.TaskConfig{}, err
 	}
 
-	for _, warning := range configSource.Delegate.Warnings() {
```

Comment:
We could probably just remove the entire `DeprecationConfigSource` type at this point. I think its entire purpose was just to print the warnings, but it was weird to have it be itself an implementation of the interface. Printing them in the task step makes a lot more sense, and with the printing removed here there's really nothing that this object does anymore. (I don't think we even use it anywhere, come to think of it.)